opennlp-tools/src/main/java/opennlp/tools/sentdetect/SentenceDetectorME.java

```diff
@@ -227,19 +243,46 @@ public Span[] sentPosDetect(String s) {
 
     for (int si = 0; si < starts.length; si++) {
       int start;
+      int wideEnd;
 
       if (si == 0) {
         start = 0;
       }
       else {
         start = starts[si - 1];
       }
+      // Define the end point of the wider span to check if the dot belongs to
+      // a date in German format
+      if (starts.length > 1 && si < starts.length - 1) {
+        wideEnd = starts[si + 1];
+      }
+      else {
+        wideEnd = s.length();
+      }
 
       // A span might contain only white spaces, in this case the length of
       // the span will be zero after trimming and should be ignored.
       Span span = new Span(start, starts[si]).trim(s);
+      // Wider span that takes into account the sentence portions before and after
+      // the split character
+      Span widerSpan = new Span(start, wideEnd).trim(s);
       if (span.length() > 0) {
-        spans[si] = span;
+        // Check if it includes a date in German format
+        if (isDateGermanFormat(widerSpan.getCoveredText(s).toString())) {
+          // If that is the case, set the span as the wider one to not split
+          // the sentence and set the position of the splitting character to
+          // avoid checking the same portion of text
+          spans[si] = widerSpan;
+          if (si == starts.length - 1) {
+            starts[si] = s.length();
+          }
+          else {
+            starts[si] = starts[si + 1];
```

Comment:
This might be easily to understand?

```suggestion
            starts[si] = Math.min(starts[si + 1], s.length());
```

Same with the one above?

Reply:
For sure, that's much better. Although now I am worried that with `starts[si + 1]`, we might get out of bounds error. Would a try-except a good solution here?

Reply:
Ah you are right please ignore me in that case as it was was better!